File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=prices[0];
        int best=0;
        for(int i=0;i<prices.size();i++){
            mini=min(mini,prices[i]);
            int profit=prices[i]-mini;
            best=max(best,profit);
        }
        return best;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution{
    public int maxArea(int[] heights){
        int l=0;
        int r=heights.length-1;
        int best=0;
        while(l<r){
            int area=(r-l)*Math.min(heights[l],heights[r]);
            best=Math.max(best,area);
            if(heights[l]<heights[r]){
                l++;
            }else{
                r--;
            }
        }
        return best;
    }
}